internal/dag/builder_test.go

```diff
@@ -4216,6 +4272,386 @@ func TestDAGInsertGatewayAPI(t *testing.T) {
 			want: listeners(),
 		},
 
+		"HTTPRoute with BackendTLSPolicy": {
+			gatewayclass: validClass,
+			gateway:      gatewayHTTPAllNamespaces,
+			upstreamTLS: &UpstreamTLS{
+				MinimumProtocolVersion: "1.2",
+				MaximumProtocolVersion: "1.2",
+			},
+			objs: []any{
+				tlsService,
+				configMapCert1,
+				&gatewayapi_v1beta1.HTTPRoute{
```

Comment:
Shall we define some helper functions to create the objects?, this files is too large now

Reply:
That makes sense. Done.